src/setupTests.ts

```diff
@@ -19,3 +22,19 @@ global.matchMedia =
       removeEventListener: jest.fn(),
     }
   }
+
+// Prevent popper from making state updates asynchronously.
+// This is necessary to avoid warnings during tests, as popper will asynchronously update state outside of test setup.
+jest.mock('@popperjs/core', () => {
+  const core = jest.requireActual('@popperjs/core')
+  return {
+    ...core,
+    createPopper: (...args: Parameters<typeof createPopper>) => {
+      const [referenceElement, popperElement, options] = args
+      if (options?.modifiers) {
+        options.modifiers = options.modifiers.filter((modifier) => !modifier.fn)
+      }
```

Comment:
is it possible that wrapping the modifiers in `act` is a better approach here? I'm not super familiar w/ popper, but I remember at one point we had modifiers with a meaningful ux impact